mlrun/model_monitoring/db/_schedules.py

```diff
@@ -13,36 +13,29 @@
 # limitations under the License.
 
 import json
+from abc import ABC
 from contextlib import AbstractContextManager
 from types import TracebackType
 from typing import Final, Optional
 
 import botocore.exceptions
 
-import mlrun.common.schemas
 import mlrun.errors
 import mlrun.model_monitoring.helpers
+from mlrun.common.schemas import ModelEndpoint
+from mlrun.common.schemas.model_monitoring.constants import ScheduleChiefFields
 from mlrun.utils import logger
 
 
-class ModelMonitoringSchedulesFile(AbstractContextManager):
+class ModelMonitoringSchedulesFileBase(AbstractContextManager, ABC):
```

Comment:
If you don't declare any abstract methods in this class - it's not truly an ABC:

```suggestion
class ModelMonitoringSchedulesFileBase(AbstractContextManager):
```